code/game/machinery/deployable.dm

```diff
@@ -53,15 +56,25 @@
 		if(!anchored)
 			return TRUE
 		var/obj/item/projectile/proj = mover
+		if(directional_blockage)
+			if(turn(mover.dir, 180) in directional_list)
+				return FALSE
 		if(proj.firer && Adjacent(proj.firer))
 			return TRUE
 		if(prob(proj_pass_rate))
 			return TRUE
 		return FALSE
-	else
-		return !density
-
-
+	if(isitem(mover)) //thrown items with the dropwall
+		if(directional_blockage)
+			if(direction_check(turn(mover.dir, 180)))
```

Comment:
This can be replaced with `if(turn(mover.dir, 180) in directional_list)` as well.

Reply:
Replaced